cmd/pipeline-template/plan_test.go

```diff
@@ -223,11 +223,11 @@ const testPlanConfig = `
             "enabled": true,
             "pubsubSystem": "google",
             "subscription": "super-derp",
-            "subscription": "super-derp",
             "subscriptionName": "super-derp",
             "source": "jack",
             "attributeConstraints": {},
-            "payloadConstraints": {}
+            "payloadConstraints": {},
+						"invalid-key": "whatever"
```

Comment:
Curious about the reason for this change? This implies that perhaps the refactoring causes some subtle change in input parsing behavior... can you explain what that change is?

Reply:
> Tests were broken after I enabled **strict parsing**, so I had to modify the pipeline-template plan tests so stop using a duplicate map key to test an invalid config. I changed it to an unknown key instead. After this, I all the tests passed, using the yaml parser.

Reply:
So enabling strict parsing could mean that some pipelines that previously worked fine for users will no longer work, correct? I suspect it's not a huge deal, but we should probably add something to the release notes if that's the case...

Reply:
The API should still error on save if the input is invalid. Ideally, this just moves the error from the server side to the client side.

That said, the Spinnaker API is pretty bad at validating content, IME. So it's possible this causes some hard errors that were previously being ignored.